stravalib/model.py

```diff
@@ -856,11 +1170,11 @@ class BaseActivityZone(
     A collection of :class:`stravalib.model.DistributionBucket` objects.
     """
 
-    # field overrides from superclass for type extensions:
-    distribution_buckets: Optional[List[DistributionBucket]] = None
+    # Field overrides from superclass for type extensions:
```

Comment:
Ok making a note here - i'm defaulting back to the original typing for these overrides - this is because they are what's breaking our tests right now. It expects the types associated with the override that support legacy behavior.

SO to fix this we will need to 

1. drop support for legacy behavior
2. remove the overrides and use the types associated with the new strava_model
3. this approach is going to fail for the instances where we are supporting undocumented behavior. i'm thinking maybe we want to come up with a slightly different approach to append new attrs to strava_model classes using some sort of composition or inheritance of a new class that adds those attrs .